distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/strategy/AssignmentStrategy.java

```diff
@@ -1,18 +1,30 @@
 package ru.fix.distributed.job.manager.strategy;
 
-import ru.fix.distributed.job.manager.model.distribution.JobState;
+import ru.fix.distributed.job.manager.model.JobId;
+import ru.fix.distributed.job.manager.model.WorkItem;
+import ru.fix.distributed.job.manager.model.AssignmentState;
+
+import java.util.List;
+import java.util.Map;
 
 /**
  * Job assignment strategy which could manage work pools distribution on workers
  */
 public interface AssignmentStrategy {
-
     /**
-     * Reassigns job
+     * At the beginning currentAssignment contains workers all workers from availability without work items
```

Comment:
I know that verbally we decided that our currentAssignment will be initially initialized with available worker list.  
As longer I read the code - the harder to accept this fact that strategy logic relay on this fact. 

Suggestion: lets say that currentAssignment initially EMPTY and do not contains anything. 
If you want, you can implement  helper methods inside AssignmentState. 
For example 
HashMap.get("worker4") rise an exception   but  you can make

AssignmentState.worker("worker4) that will create worker4 key in absent and return HashSet<WorkItems>

Reply:
done